testing/E2E_Test_Update.md

```diff
@@ -0,0 +1,164 @@
+are to: 
+
+* outline possible approaches to writing tests and documentation on how to implement 
+ them
+* provide an update on current state of e2e testing
+* provide documentation on Appium and tools for help in creation of e2e tests
+* provide an estimation on the time required for developing e2e tests
+
+## Supporting Documentation
+
+[Previous Test Proposal](./supporting_docs/previous_test_proposal.md)
+
+[Appium Background Information](./supporting_docs/appium_information.md) - Contains 
+information on Appium, its clients and some important considerations when selecting 
+an Appium client library. 
+
+[UI Selectors And Their Importance](./supporting_docs/ui_selectors_and_their_importance.md) - 
+Contains information on UI selectors, their impact on a UI test suite, and examples 
+of how to implement some preferred selectors in mobile applications.  
+
+[Tools for Finding UI Selectors](./supporting_docs/tools_to_find_selectors.md) - 
+Contains information on tools that can be used to identify selectors and generate
+unrefined e2e test code.
+
+## The Previous Test Proposal
+
+The previous test proposal sections on test suites and test infrastructure can be
+found at the following links:
+
+[Proposal on Test Suite](./e2e_drafts/)
+
+[Proposal on Test Infrastructure](./e2e_drafts/E2E_Test_infrastructure.md)
+
+### Test Suite Proposal Summary
+The main criteria that were used to create the proposal for the test suite 
+for end to end UI tests, were:
+
+* Tests must be readable to enable incoming developers to get involved 
+quickly in the maintenance of the test suite and production of new tests.
+
+* The test suite should be easily maintainable to reduce the time spent on 
+fixing bugs and the amount of developers required to maintain it.
+
+* In order to fulfill the required test cases, the framework used to create 
+the test suite needs to be able to handle interactions from applications 
+that redirects it to the browser for signing into the application.
+
+During the research phase, various test frameworks for android apps were 
+investigated including UIAutomator, Espresso and Appium including multiple 
+appium clients (some information in Appium Clients section).  It was 
+determined, from this research, that the only framework capable of performing 
+application and browser redirect actions was Appium, and therefore it should 
+be used for end to end UI testing. 
+
+In order to facilitate the other two expectations of the test suite, it was 
+proposed that best practices such as page objects should be used to avoid 
+code repetition and split the presentation logic from the test logic.  While 
+the Appium Java-client was the preferred option to create the tests, the 
+proposal called for using Kotlin to create a domain specific language (DSL) 
+to write the tests.  This decision was made as this approach would provide 
+easier readability and understanding for new developers.
```

Comment:
There's an argument for Kotlin, as this is replacement for Java in Android development, that is a skill we should master in the mobile team, so we can try it in some safe way, not yet including it in SDK for example. :)

Reply:
A couple of thoughts here. 

@matskiv it is not Kotlin that will be used to write tests its a domain specific language created from Kotlin. It could be even more English based than a C-like syntax making it even more readable.

I this case the library that you will write tests is made with Kotlin. As @wojta has said there is an advantage in using Kotlin for something like this as it can introduce people to Kotlin based Android development which was one of the thoughts in the original proposal.

Reply:
@twnolan Thank you for the explanation.

IMHO: custom DSL seem unnecessary. I am in favor of more broadly used language + page objects(or similar concept).
FYI: On a recent weekly call we discussed this topic and agreed to spike Java and JS based clients for Appium.